src/components/Researcher/ParticipantList/PatientStudyCreator.tsx

```diff
@@ -290,6 +300,40 @@ export default function PatientStudyCreator({
     })
   }
 
+  const showResult = (id: string) => {
+    enqueueSnackbar(
+      t("Successfully created Participant id. Tap the expand icon on the right to see credentials and details.", {
+        id: id,
+      }),
+      {
+        variant: "success",
+        persist: true,
+        content: (key: string, message: string) => (
+          <SnackMessage id={key} message={message}>
+            <TextField variant="outlined" size="small" label={t("Temporary email address")} value={`${id}@lamp.com`} />
+            <Box style={{ height: 16 }} />
+            <TextField variant="outlined" size="small" label={t("Temporary password")} value={`${id}`} />
+            <Grid item>
+              <TextField
+                fullWidth
+                label={t("One-time login link")}
+                style={{ marginTop: 16 }}
+                variant="outlined"
+                value={_qrLink(`${id}@lamp.com`, id)}
+                onChange={(event) => {}}
+              />
+              <Tooltip title={t("Scan this QR code on a mobile device to automatically open a user dashboard.")}>
+                <Grid container justify="center" style={{ padding: 16 }}>
+                  <QRCode size={256} level="H" value={_qrLink(`${id}@lamp.com`, id)} />
+                </Grid>
+              </Tooltip>
+            </Grid>
+          </SnackMessage>
+        ),
+      }
+    )
+  }
+
```

Comment:
[Why is this code duplicated from `AddUser.tsx`?](https://github.com/BIDMCDigitalPsychiatry/LAMP-dashboard/blob/007020c0bfa1423776f18ab67ac9403b014b0943/src/components/Researcher/ParticipantList/AddUser.tsx#L131) Code that is re-used in more than one place should always be refactored into a separate file and imported. Please change this.

Reply:
@avaidyam As this is enqueueSnackbar, I was not able to make it as separate component. I tried to split the SnackMessage part from different component , but it was showing error " Cannot read property 'getBoundingClientRect' of null".

Reply:
You should be able to make it a separate function (not a React component) in a separate file though?